tools/cli/commands/connect.py

```diff
@@ -0,0 +1,263 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for implementing the `datalab connect` command."""
+
+import argparse
+import subprocess
+import tempfile
+import threading
+import urllib2
+import webbrowser
+
+
+description=(
+"""{0} {1} creates a persistent connection to a
+Datalab instance running in a Google Compute Engine VM.
+
+This is a thin wrapper around the *ssh(1)* command that takes care
+of authentication and the translation of the instance name into an
+IP address.
+
+This command ensures that the user's public SSH key is present
+in the project's metadata. If the user does not have a public
+SSH key, one is generated using *ssh-keygen(1)* (if the --quiet
+flag is given, the generated key will have an empty passphrase).
+
+Once the connection is established, this command opens a browser
+window pointing at Datalab. To disable the browser opening, add
+the --no-launch-browser flag.
+
+This command will attempt to re-establish the connection if it
+gets dropped. However, that connection will only exist while
+this command is running.
+""")
+
+
+examples=(
+"""
+To connect to 'example-instance' in zone 'us-central1-a', run:
+
+    $ {0} {1} example-instance --zone us-central1-a"""
```

Comment:
I'm worried about drifting too far from the conventions of the existing command line tools.

As far as I am aware, none of the existing tools (e.g. gcloud, gsutil, etc) support referring to instances in this way.